src/Framework/AssemblyLoadBuildEventArgs.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+
+#nullable disable
+
+using System;
+using System.IO;
+using Microsoft.Build.Shared;
+
+namespace Microsoft.Build.Framework
+{
+    public class AssemblyLoadBuildEventArgs : BuildMessageEventArgs
+    {
+        public AssemblyLoadBuildEventArgs()
+        { }
+
+        public AssemblyLoadBuildEventArgs(
+            AssemblyLoadingContext loadingContext,
+            string assemblyName,
+            string assemblyPath,
+            Guid mvid,
+            int appDomainId,
+            string appDomainFriendlyName,
+            MessageImportance importance = MessageImportance.Low)
+            : base(null, null, null, importance, DateTime.UtcNow, assemblyName, assemblyPath, mvid)
+        {
+            LoadingContext = loadingContext;
+            AssemblyName = assemblyName;
+            AssemblyPath = assemblyPath;
+            MVID = mvid;
+            AppDomainId = appDomainId;
+            AppDomainFriendlyName = appDomainFriendlyName;
+        }
+
+        public AssemblyLoadingContext LoadingContext { get; private set; }
```

Comment:
This is a great representation for the task case, where the event context information is sufficient to understand what task did the load. In the other cases, it might be nice to have an association to "what plugin did the loading?". Only way I can see out of that is a string.